index.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+def bubble_sort(arr)
+  k = arr.size
+  count = 0
+  k.times do |i|
+    n = 0
+    while n < (arr.size - 1 - i)
+      count += 1
+      arr[n], arr[n + 1] = arr[n + 1], arr[n] if arr[n] > arr[n + 1]
+      n += 1
+    end
+  end
+  arr
+end
+
+bubble_sort([9, 7, 6, 5, 4, 3, 2, 1])
+
+def bubble_sort_by(arr)
+  k = arr.size
+  count = 0
+  k.times do |i|
+    n = 0
+    while n < (arr.size - 1 - i)
+      count += 1
+      raise 'No block given. Please supply a block.' unless block_given?
```

Comment:
It's not the good place.
Putting it here mean that you're going to test it every times in the while loop.
Think about put it before the starting the while loop or after ```def bubble_sort_by(arr)``` ?